compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2240,6 +2240,182 @@ OMR::Power::TreeEvaluator::lbitpermuteEvaluator(TR::Node *node, TR::CodeGenerato
    return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
    }
 
+static inline TR::Register*
+compressbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg, TR::DataType type)
+   {
+   TR::Node *srcNode = node->getFirstChild();
+   TR::Node *maskNode = node->getSecondChild();
+   TR::Register *srcReg = cg->evaluate(srcNode);
+   TR::Register *maskReg = NULL;
+   TR::Register *resReg = cg->allocateRegister();
+
+   if (type != TR::Int64)
+      {
+      TR::InstOpCode::Mnemonic loadOpcode;
+      uint64_t maskConst;
+      uint32_t length;
+      switch (type)
+         {
+         case TR::Int32:
+            loadOpcode = TR::InstOpCode::lwz;
+            maskConst = CONSTANT64(0x00000000ffffffff);
+            length = 4;
+            break;
+         case TR::Int16:
+            loadOpcode = TR::InstOpCode::lhz;
+            maskConst = CONSTANT64(0x000000000000ffff);
+            length = 2;
+            break;
+         case TR::Int8:
+            loadOpcode = TR::InstOpCode::lbz;
+            maskConst = CONSTANT64(0x00000000000000ff);
+            length = 1;
+            break;
+         default:
+            TR_ASSERT_FATAL(false, "Unrecognized compressbits type %s\n", type.toString());
+            break;
+         }
+
```

Comment:
since `lbz/lhz/lwz`  meant zero-extension already (with resulting register upper bits cleared), i don't see why you cannot simplify the logic below to handle it consistently, i.e. you only need `rldicl` for cases where mask register pre-exists and types are not 64bit.